### ISSUE TITLE
Neo4j 2.2.0 bug querying Lucene

### ISSUE THREAD

**Issue by riripi** [neo4j/neo4j#4444]
I am passing from Neo4j 1.9.9 to Neo4j 2.2.0, with embedded mode using java. I have inserted the transactions also for read operations, but now when I query my Lucene indexes as this

```
rhits = index.query("cs", "*");
out.println("#" + rhits.size());
rhits.close();
```

as you can see I do not have to iterate over the result, I need only the number of results but this new Neo4j version looks like loading all in memory and I get the following error in the first instruction.

```
Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
        at org.neo4j.collection.primitive.hopscotch.IntArrayBasedKeyTable.initia
lizeTable(IntArrayBasedKeyTable.java:54)
        at org.neo4j.collection.primitive.hopscotch.IntArrayBasedKeyTable.<init>
(IntArrayBasedKeyTable.java:48)
        at org.neo4j.collection.primitive.hopscotch.LongKeyTable.<init>(LongKeyT
able.java:27)
        at org.neo4j.collection.primitive.Primitive.longSet(Primitive.java:66)
        at org.neo4j.kernel.impl.coreapi.LegacyIndexProxy$1.<init>(LegacyIndexPr
oxy.java:296)
        at org.neo4j.kernel.impl.coreapi.LegacyIndexProxy.wrapIndexHits(LegacyIn
dexProxy.java:294)
        at org.neo4j.kernel.impl.coreapi.LegacyIndexProxy.query(LegacyIndexProxy
.java:352)
```

I never get this with older versions of Neo4j! I always did this operation until version 1.9.9.

**Comment by tinwelint** [comment]
See https://github.com/neo4j/neo4j/pull/4509 for a solution to this memory issue.

**Comment by riripi** [comment]
Thank you very much! When will be possibile to have this solution into a new release?

**Comment by riripi** [comment]
Solved in version 2.2.2. Thank you!
